packages/jest/src/automatic.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Options for Automatic checks to be passed to {@link registerSa11yAutomaticChecks}
+ */
+export type AutoCheckOpts = {
+    runAfterEach?: boolean;
+    cleanupAfterEach?: boolean;
+    excludeTests?: string[];
+};
+
+/**
+ * Default options when {@link registerSa11yAutomaticChecks} is invoked
+ */
+const defaultAutoCheckOpts: AutoCheckOpts = {
+    runAfterEach: true,
+    cleanupAfterEach: true,
+    excludeTests: [],
+};
+
+/**
+ * Run accessibility check on each node in the DOM using {@link toBeAccessible}
+ * @param cleanup - boolean indicating if the DOM should be cleaned up after
+ */
+export async function automaticCheck(cleanup = false): Promise<void> {
+    while (document.body.firstChild) {
+        try {
+            // TODO (refactor): check if sa11y matcher is registered and register?
+            await expect(document.body.firstChild).toBeAccessible();
+        } finally {
+            // TODO (spike): Is there a better way to walk the DOM that makes cleanup optional ?
```

Comment:
How about something like the following?

```
let curr = document.body.firstChild
while (curr !== null) {
  await expect(curr).toBeAccessible();
  curr = curr.nextSibling
}
```

Reply:
Thanks @trevor-bliss

Reply:
Tried out your suggestion, iterated on it and ended up with the current version. Please review and let me know.